src/Control/Distributed/Process/Node.hs

```diff
@@ -319,11 +322,38 @@ startServiceProcesses node = do
 
 -- | Force-close a local node
 --
--- TODO: for now we just close the associated endpoint
+-- This function first try to brutally kill all processes, this
+-- process is not guaranteed to continue. So this is application
+-- responsibility to either close all it's processes before dying
+-- or being able to gracefully shutdown.
 closeLocalNode :: LocalNode -> IO ()
-closeLocalNode node =
-  -- TODO: close all our processes, surely!?
-  NT.closeEndPoint (localEndPoint node)
+closeLocalNode node = join $ terminateProcesses
+  where
+    terminateProcesses :: IO (IO ())
```

Comment:
Good point about new processes, but the current solution is insufficient in addition to being dangerous: the fact that there are no processes any longer on the node does not mean that a process on another node can't spawn a process on this node before the node is actually shut down.

The proper fix in my mind is neither of the 3 solutions above: clearly nodes need one additional state, say a "closing" state. When a node is in the closing state, no process is allowed to assign any new resources (e.g. newly spawned processes) to it.